### PR TITLE
Make two small changes and add regression tests for dispatcher

### PR DESCRIPTION
**Pull request by eurekaka**

### CODE REVIEW

File: src/backend/tcop/postgres.c [hunk 1]
```diff
@@ -4634,7 +4634,12 @@ PostgresMain(int argc, char *argv[],
 
 	/* Also send GPDB QE-backend startup info (motion listener, version). */
 	if (Gp_role == GP_ROLE_EXECUTE)
-		sendQEDetails();
+	{
+#ifdef FAULT_INJECTOR
```
Comment (asimrp) [inline src/backend/tcop/postgres.c]: You don't need the `#ifdef` around `SIMPLE_FAULT_INJECTOR()` macro.  It's already defined to be a no-op if fault injectors are disabled.

Comment (eurekaka) [inline src/backend/tcop/postgres.c]: If simply using SIMPLE_FAULT_INJECTOR() I think it's OK to remove the #ifdef, but we have a check of the return value here, so the source code may looks like
```
if ( != FaultInjectorTypeSkip)
```
after preprocessing if FAULT_INJECTOR is not defined.

Comment (asimrp) [inline src/backend/tcop/postgres.c]: Sorry, I overlooked that `if()`.  My bad.

**Comment by eurekaka** [comment]
Merged, thanks for reviewing.

**Pull request closed without merging**